tests/test_in_toto_run.py

```diff
@@ -129,6 +128,88 @@ def test_main_optional_args(self):
     self.assertListEqual(list(link_metadata.signed.products.keys()),
         [self.test_artifact[len(strip_prefix):]])
 
+  def test_main_exclude_patterns(self):
+    """Test CLI command with edge cases exclude patterns."""
+    # Hardcoded sha256 of an empty file.
+    sha256 = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+
+    # Create tmp directories and artifacts for pattern test cases.
+    Path("artifacts").mkdir()
+    Path("artifacts/foo").mkdir()
+    Path("artifacts/foo/test_pkgA.xyz").touch()
+    Path("artifacts/foo/test_pkgB.xyz").touch()
+    Path("artifacts/foo/bar").mkdir()
+    Path("artifacts/foo/bar/test_pkgA.xyz").touch()
+    Path("artifacts/foo/bar/test_pkgB.xyz").touch()
+    Path("#foo#").touch()
+    Path("#bar!").touch()
+    Path("frotz").mkdir()
+    Path("frotz/test_artifact").touch()
+    Path("doc").mkdir()
+    Path("doc/frotz").mkdir()
+    Path("doc/frotz/test_artifact").touch()
+    Path("a").mkdir()
+    Path("a/doc").mkdir()
+    Path("a/doc/frotz").mkdir()
+    Path("a/doc/frotz/test_artifact").touch()
+
+    # Define exclude pattern test cases. Each test case is a tuple with three
+    # values: items, pattern, and expected result.
+    test_cases = [
+      (self.test_artifact, "/**/test_artifact", {}),
```

Comment:
Can we add a test case for:

```A single / does not match any file.```

Reply:
From my tests `/` matched every file. But maybe I did something wrong, or it's a defect.

![image](https://user-images.githubusercontent.com/10353018/154359744-ebeaf869-094c-4b99-98cd-de1185279980.png)

Reply:
I have misled myself :man_facepalming: :man_facepalming: :man_facepalming:. 

In the initial exclude test cases implementation I was asserting "opposite" values: 
Files that will be matched by exclude pattern VS files that will remain, hence the confusion. In reality, it works properly and I just was asserting the opposite expected output.